modules/readline/8.2/overlay/BUILD.bazel

```diff
@@ -0,0 +1,132 @@
+        "#undef HAVE_POSIX_SIGSETJMP": "#define HAVE_POSIX_SIGSETJMP 1",
+        "#undef HAVE_PSELECT": "#define HAVE_PSELECT 1",
+        "#undef HAVE_PUTENV": "#define HAVE_PUTENV 1",
+        "#undef HAVE_PWD_H": "#define HAVE_PWD_H 1",
+        "#undef HAVE_READLINK": "#define HAVE_READLINK 1",
+        "#undef HAVE_SELECT": "#define HAVE_SELECT 1",
+        "#undef HAVE_SETENV": "#define HAVE_SETENV 1",
+        "#undef HAVE_SETLOCALE": "#define HAVE_SETLOCALE 1",
+        "#undef HAVE_STDARG_H": "#define HAVE_STDARG_H 1",
+        "#undef HAVE_STDBOOL_H": "#define HAVE_STDBOOL_H 1",
+        "#undef HAVE_STDLIB_H": "#define HAVE_STDLIB_H 1",
+        "#undef HAVE_STRCASECMP": "#define HAVE_STRCASECMP 1",
+        "#undef HAVE_STRCOLL": "#define HAVE_STRCOLL 1",
+        "#undef HAVE_STRINGS_H": "#define HAVE_STRINGS_H 1",
+        "#undef HAVE_STRING_H": "#define HAVE_STRING_H 1",
+        "#undef HAVE_STRPBRK": "#define HAVE_STRPBRK 1",
+        "#undef HAVE_STRUCT_DIRENT_D_FILENO": "#define HAVE_STRUCT_DIRENT_D_FILENO 1",
+        "#undef HAVE_STRUCT_DIRENT_D_INO": "#define HAVE_STRUCT_DIRENT_D_INO 1",
+        "#undef HAVE_SYS_FILE_H": "#define HAVE_SYS_FILE_H 1",
+        "#undef HAVE_SYS_IOCTL_H": "#define HAVE_SYS_IOCTL_H 1",
+        "#undef HAVE_SYS_SELECT_H": "#define HAVE_SYS_SELECT_H 1",
+        "#undef HAVE_TCGETATTR": "#define HAVE_TCGETATTR 1",
+        "#undef HAVE_TERMCAP_H": "#define HAVE_TERMCAP_H 1",
+        "#undef HAVE_TERMIOS_H": "#define HAVE_TERMIOS_H 1",
+        "#undef HAVE_TERMIO_H": "#define HAVE_TERMIO_H 1",
+        "#undef HAVE_TOWLOWER": "#define HAVE_TOWLOWER 1",
+        "#undef HAVE_TOWUPPER": "#define HAVE_TOWUPPER 1",
+        "#undef HAVE_UNISTD_H": "#define HAVE_UNISTD_H 1",
+        "#undef HAVE_VSNPRINTF": "#define HAVE_VSNPRINTF 1",
+        "#undef HAVE_WCHAR_H": "#define HAVE_WCHAR_H 1",
+        "#undef HAVE_WCHAR_T": "#define HAVE_WCHAR_T 1",
+        "#undef HAVE_WCRTOMB": "#define HAVE_WCRTOMB 1",
+        "#undef HAVE_WCSCOLL": "#define HAVE_WCSCOLL 1",
+        "#undef HAVE_WCTYPE_H": "#define HAVE_WCTYPE_H 1",
+        "#undef HAVE_WCTYPE_T": "#define HAVE_WCTYPE_T 1",
+        "#undef HAVE_WCWIDTH": "#define HAVE_WCWIDTH 1",
+        "#undef HAVE_WINT_T": "#define HAVE_WINT_T 1",
+        "#undef PROTOTYPES": "#define PROTOTYPES 1",
+        "#undef RETSIGTYPE": "#define RETSIGTYPE void",
+        "#undef STDC_HEADERS": "#define STDC_HEADERS 1",
+        "#undef STRUCT_WINSIZE_IN_SYS_IOCTL": "#define STRUCT_WINSIZE_IN_SYS_IOCTL 1",
+        "#undef VOID_SIGHANDLER": "#define VOID_SIGHANDLER 1",
+        "#undef _ALL_SOURCE": "#define _ALL_SOURCE 1",
+        "#undef _GNU_SOURCE": "#define _GNU_SOURCE 1",
+        "#undef _POSIX_PTHREAD_SEMANTICS": "#define _POSIX_PTHREAD_SEMANTICS 1",
+        "#undef _TANDEM_SOURCE": "#define _TANDEM_SOURCE 1",
+        "#undef __EXTENSIONS__": "#define __EXTENSIONS__ 1",
+        "#undef __PROTOTYPES": "#define __PROTOTYPES 1",
+    } | {
```

Comment:
This seems unnecessary?

Reply:
It’s to ensure ordering while not getting impacted by buildifier’s dict sorting. Replacing `FOO` before `FOO_BAR` breaks the latter since `FOO` would have already been replaced.